### ISSUE TITLE
watchOS 4 support?

### ISSUE THREAD

**Issue by kudit** [apple/swift-collections#407]
I read somewhere that this package has no minimum requirements however when I tried to include in a package that supports watchOS 4, Xcode generates warnings that this package requires watchOS 5.

### Information

- **1.1.2:** What tag or branch of swift-collections are you using?
- **macOS 14.5:** Please tell us the version number of your operating system.
- **5.10:** Paste the output of `swift --version` here.
- Xcode Version 15.4 (15F31d)

### Checklist
- [x] If possible, I've reproduced the issue using the `main` branch of this package.
- [x] I've searched for [existing GitHub issues](https://github.com/apple/swift-collections/issues).

### Steps to Reproduce
Create a swift package that has a minimum requirement of watchOS 4 (like https://github.com/kudit/Compatibility)
Attempt to build Watch App target.

### Expected behavior
inclusion is fine and compiles in Xcode with no errors.

### Actual behavior
Xcode reports the following error:
`The package product 'OrderedCollections' requires minimum platform version 5.0 for the watchOS platform, but this target supports 4.0`

**Comment by lorentey** [comment]
This package does not intentionally set a minimum deployment target.

Do you happen to know where the watchOS 5 minimum comes from?

**Comment by glessard** [comment]
The description is about creating a swift package, and I can't reproduce this with a swift package with Xcode 15.4.

However I am seeing something similar but not identical, using the same version of Xcode. I opened a new watchOS project for a library, then added swift-collections (1.1.3). I set the watchOS deployment target to the lowest available (watchOS 4). Upon building, I see:
```
ld: warning: object file (<path>/Build/Products/Debug-watchos/OrderedCollections.o) was built for newer watchOS version (5.0) than being linked (4.0)
ld: warning: object file (<path>/Build/Products/Debug-watchos/InternalCollectionsUtilities.o) was built for newer watchOS version (5.0) than being linked (4.0)
```
For what it's worth with no change in project settings, this does not happen in the current Xcode 16 beta.
